cpp/src/wholememory_ops/functions/gather_scatter_func.cuh

```diff
@@ -365,6 +366,8 @@ void gather_temp_func(wholememory_gref_t embedding_gref,
   }
   int block_size  = 1024;
   int block_count = indice_count > 1568 ? 1568 : indice_count;
+  if (gather_sms != -1) block_count = gather_sms;
+  // printf("the gather kernel sm number is %d\n", block_count);
   kernel_fn<<<block_count, block_size, 0, stream>>>(embedding_gref,
```

Comment:
if(gather_sms !=-1){
    ... check gather_sms , if (gather_sms<0 || gather_sms> some_max_value){
        error or warning
    }
}

Reply:
Fixed by checking if the gather_sms is valid in the "create_embedding" func.